lib/perl/Genome/Model/ReferenceAlignment/Command/LibraryDuplicationRates.pm

```diff
@@ -0,0 +1,86 @@
+package Genome::Model::ReferenceAlignment::Command::LibraryDuplicationRates;
+
+use strict;
+use warnings;
+
+class Genome::Model::ReferenceAlignment::Command::LibraryDuplicationRates {
+    is => 'Command::V2',
+    has => [
+        builds => {
+            is_many => 1,
+            is => 'Genome::Model::Build::ReferenceAlignment',
+            shell_args_position => 1,
+            doc => 'List of builds to report on.',
+        },
+    ],
+};
+
+sub help_brief {
+    'A command to print Picard MarkDuplicates duplication metrics for a reference alignment build.'
+}
+
+sub help_synopsis {
+    'Dump reference alignment build duplication metrics to standard out in tab-delimited format.'
+}
+
+sub help_detail{
+    return <<"EOS"
+All reference alignment builds currently use Picard MarkDuplicates to flag PCR duplicates after merging the individual instrument data alignments.
+One output from Picard MarkDuplicates is a metrics file that provides a per library breakdown of PCR and optical duplicate reads.  This tool simply parses the metrics file and dumps those metrics along with basic build/model information in tab-delimited form.
+EOS
+}
+
+sub execute {
+    my $self = shift;
+
+    my @data;
+    for my $build ($self->builds) {
+        my @instrument_data = $build->instrument_data;
+        my @library_names = List::MoreUtils::uniq(sort map { $_->library->name } @instrument_data);
+
+        for my $library_name (@library_names) {
+            my @library_instrument_data = grep {$_->library_name eq $library_name} @instrument_data;
+            my $data = {
+                'BUILD_ID' => $build->id,
+                'MODEL_ID' => $build->model->id,
+                'SUBJECT_NAME' => $build->model->subject->name,
+                'COUNT_INSTRUMENT_DATA' => scalar(@library_instrument_data),
```

Comment:
If `@library_instrument_data` is used only to get this count, we could do something like this:
```perl
my %library_counts;
for my $instrument_data (@instrument_data) {
  $library_counts{$instrument_data->library_name}++;
}
```
and then not need to `grep` through the instrument data again each time through the library name loop.